h2o-r/h2o-package/R/frame.R

```diff
@@ -1155,31 +1155,51 @@ NULL
 
   # This function is called with a huge variety of argument styles
   # Here's the breakdown:
-  #   Style          Type #args  Description
-  # df[]           - na na 2    both missing, identity with df
-  # df["colname"]  - c  na 2    single column by name, df$colname
-  # df[3]          - X  na 2    if ncol > 1 then column else row
-  # df[,]          - na na 3    both missing, identity with df
-  # df[2,]         - r  na 3    constant row, all cols
-  # df[1:150,]     - r  na 3    selection of rows, all cols
-  # df[,3]         - na c  3    constant column
-  # df[,1:10]      - na c  3    selection of columns
-  # df[,"colname"] - na c  3    single column by name
-  # df[2,"colname"]- r  c  3    row slice and column-by-name
-  # df[2,3]        - r  c  3    single element
-  # df[1:150,1:10] - r  c  3    rectangular slice
-  # df[a<b,]       - f  na 3    boolean row slice
-  # df[a<b,c]      - f  c  3    boolean row slice
-  is1by1 <- !missing(col) && !missing(row) && !is.H2OFrame(row) && length(col) == 1 && length(row) == 1
+  #   Style          Type  #args  Description
+  # df[]             - na na 2    both missing, identity with df
+  # df["colname"]    - c  na 2    single column by name, df$colname
+  # df[3]            - X  na 2    if ncol > 1 then column else row
+  # df[,]            - na na 3    both missing, identity with df
+  # df[2,]           - r  na 3    constant row, all cols
+  # df[1:150,]       - r  na 3    selection of rows, all cols
+  # df[,3]           - na c  3    constant column
+  # df[,1:10]        - na c  3    selection of columns
+  # df[,"colname"]   - na c  3    single column by name
+  # df[2,"colname"]  - r  c  3    row slice and column-by-name
+  # df[2,3]          - r  c  3    single element
+  # df[1:150,1:10]   - r  c  3    rectangular slice
+  # df[a<b,]         - f  na 3    boolean row slice
+  # df[a<b,c]        - f  c  3    boolean row slice
+  # df[1,-1]         - r  c  3    selection of first row minus the first column
+  # df[-1,-1]        - r  c  3    get rid of first row and first column
+  # df[-1,1]         - r  c  3    get rid of first row and keep first column
+  # df[-1:-20,-1:-3] - r  c  3    get rid of first 20 rows and first 3 columns
+  # df[-1:-20,1:3]   - r  c  3    get rid of first 20 rows and keep first 3 columns
+  # df[1:20,-1:-3]   - r  c  3    keep first 20 rows and remove first 3 columns
+
+  # Boolean check for negative indexes
+  is_neg_idx <- FALSE
+  # Have a row & column selector with negative col or negative row indexes?
+  if(!missing(col) && !missing(row) && !is.H2OFrame(row) && !is.H2OFrame(col) && (col < 0 || row < 0)){
```

Comment:
Ah, yes. They do support that, i.e., 

`n = c(2, 3, 5) 
s = c("aa", "bb", "cc") 
b = c(TRUE, FALSE, TRUE) 
df = data.frame(n, s, b) 
> all(df[,0:-1] == df[,-1])
[1] TRUE` 

So we can do `&& (col <= 0 || row <= 0)`